// jest-cli/build/generate_empty_coverage.js
'use strict';var _istanbulLibInstrument;











function _load_istanbulLibInstrument() {return _istanbulLibInstrument = require('istanbul-lib-instrument');}var _jestRuntime;
function _load_jestRuntime() {return _jestRuntime = _interopRequireDefault(require('jest-runtime'));}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

module.exports = function (
source,
filename,
globalConfig,
config)
{
  const coverageOptions = {
    collectCoverage: globalConfig.collectCoverage,
    collectCoverageFrom: globalConfig.collectCoverageFrom,
    collectCoverageOnlyFrom: globalConfig.collectCoverageOnlyFrom,
    mapCoverage: globalConfig.mapCoverage };

  if ((_jestRuntime || _load_jestRuntime()).default.shouldInstrument(filename, coverageOptions, config)) {
    // Transform file without instrumentation first, to make sure produced
    // source code is ES6 (no flowtypes etc.) and can be instrumented
    const transformResult = new (_jestRuntime || _load_jestRuntime()).default.ScriptTransformer(
    config).
    transformSource(filename, source, false, globalConfig.mapCoverage);
    const instrumenter = (0, (_istanbulLibInstrument || _load_istanbulLibInstrument()).createInstrumenter)();
    instrumenter.instrumentSync(transformResult.code, filename);
    return {
      coverage: instrumenter.fileCoverage,
      sourceMapPath: transformResult.sourceMapPath };

  } else {
    return null;
  }
}; /**
    * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
    *
    * This source code is licensed under the BSD-style license found in the
    * LICENSE file in the root directory of this source tree. An additional grant
    * of patent rights can be found in the PATENTS file in the same directory.
    *
    * 
    */
